postgres/osmap.yaml

```diff
@@ -91,9 +91,45 @@ RedHat:
 {% endif %}
 
 Suse:
+  pkg_repo:
+    name: pgdg-sles-{{ release }}
+    humanname: PostgreSQL {{ repo.version }} $releasever - $basearch
+    #Using sles-12 upstream repo for opensuse
+    baseurl: 'https://download.postgresql.org/pub/repos/zypp/{{ repo.version }}/suse/sles-12-$basearch'
+    key_url: 'https://download.postgresql.org/pub/repos/zypp/{{ repo.version }}/suse/sles-12-$basearch/repodata/repomd.xml.key'
+    gpgcheck: 1
+    gpgautoimport: True
+
+{% if repo.use_upstream_repo %}
+  {# Pillars needed are 'use_upstream_repo: True' and 'version: n.n'. #}
+  {# Avoid setting package names as pillars, as may corrupt postgres. #}
+  {% set lib_dir = '/var/lib/pgsql/' ~ repo.version ~ '/data' %}
+
+  pkg: postgresql{{ release }}-server
+  pkg_client: postgresql{{ release }}
+  conf_dir: {{ lib_dir }}
+  service: postgresql-{{ repo.version }}
+
+  #This is postgresql-libs in defaults.yml but upstream is libpqxx
+  pkg_libpq_dev: libpqxx
+  pkg_dev: postgresql{{ release }}-devel
+  pkg_libs: postgresql{{ release }}-libs
+  pkg_contrib: postgresql{{ release }}-contrib
+
+  prepare_cluster:
+    #Full path needed as initdb is NOT 'cross version compatible' binary
+    command: /usr/pgsql-{{ repo.version }}/bin/initdb --pgdata='{{ lib_dir }}'
+    test: test -f '{{ lib_dir }}/PG_VERSION'
+
+{% else %}
+
   pkg: postgresql-server
   pkg_client: postgresql
-  pkg_libpq_dev: postgresql
+  pkg_libpq_dev: libpq5
+  pkg_libs: postgresql-libs
+  pkg_contrib: postgresql-contrib
```

Comment:
You may put the `postgresql-contrib` back into the proper section in `pillar.example` file with comment saying this package could be a requirement for installing some extensions.

Reply:
I don't want to. Pillar handling exposes production users to unacceptable risk.

- Handling packages via osmap.yaml pattern is good.
- Handling packages **only** via pillars (i.e. postgresql_contrib) is bad pattern.
-- Missing Postgresql-contrib package causes bug #172.
-- Specifying wrong contrib_pkg pillar corrupts package manager, the PG installation, risks data loss. The PG schema may get upgraded too. This is all bad.
- And trying to handle #172 via comments is wrong approach.

Reply:
I mostly agree with all your points.
My main point is that `pkg_contrib` have not been used within current state of your changes, so having this in defaults and OS mapping files is meaningless right now. It just does nothing.
And this does not really help to solve issue #172 taking into an account broad scope of distributions supported by the formula.
I will try to put my thoughts there. Let's concentrate on bringing in initial basic support for SUSE systems (maybe less functional), but without breaking things which already work.

Reply:
I'll remove these entries from osmap.yaml but don't want pkg_contrib in pillar.example. Apart from the reasons above, I've done lots of verification on this formula past few weeks across many scenarios and I just don't agree with packages in pillars anymore - there are too many use cases.  But, I do want the PR merged, so tell me if you want pillar.example restored.